### PR TITLE
DOCS/Contributing

### PR DESCRIPTION
**Pull request by Dwaynekj**
@joneit @rdepena @nkolba

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Welcome
+
+Third-party contributors are essential for keeping Hypergrid great. We simply cannot imagine
```
Comment (rdepena) [inline CONTRIBUTING.md]: not sure we need this paragraph.

Comment (Dwaynekj) [inline CONTRIBUTING.md]: Just trying to be nice and friendly

Comment (rdepena) [inline CONTRIBUTING.md]: nah, lets stick with facts.

**Comment by Dwaynekj** [comment]
@joneit Any feedback?

**Comment by joneit** [comment]
* "apart" is in this usage two words: "a part"
* why is `help wanted` in code font?
* "set of repos" needs period appended
* change "squash your commits, if needs, be to facilitate that" to "squash your commits as needed to facilitate that"
* after "rebase your work on top of develop" add "as needed"
* change Opera to Safari
* I don't think we currently have any support for vertical alignment. (Do we?)
* Replace note on commenting code ("We are not believers ...") with: "Code should be as self-explanatory as possible by using well-considered variable names; additional variables for intermediate values instead of unexplained subexpressions; clear, logical flow; parallel structure; etc. Use comments only to explain any remaining subtleties."
* change "on the same token" to "on the other hand" and make it the leading sentence of a new paragraph
* replace the curly brace after `[tutorial]` with a parenthesis.
* change "regular basis in time" to simply "regular basis"
* work-in-progress note:
  * remove it
  * replace it with: "do not submit until ready to publish — and then hold off a bit longer until you feel certain you are not submitting prematurely." 
  * and add: "if you find you absolutely must update a pull request, you _must_ leave an explanatory comment. Updating will delay merging your PR if we have to review it again. Please try to avoid doing this (by not submitting too early)"
* change "and notes to our revision history" to ", add your submission notes to our release notes,"
* the Puppet sample commit is not so great. Also you removed the first line, an important part of a commit message. Put it back but remove ticket number: "Make the example in CONTRIBUTING imperative and concrete" and remove the part of the message body that refers to a ticket number.  You should also mention to try to keep first line < 50 chars.

Rather than this:

```
* `git clone <your fork>`
* `node -v` //=> v4.0.0
* `npm -v`  //=> 2.14.2
*  We recommend using nvm to manage your versions of node and npm
* `npm install -g gulp`
* `npm install`
* `gulp`
```

make it a bash session (note leading prompt string $ + space and comment char #):

```
```bash
$ git clone <your fork>
$ node -v # at least:
v4.0.0
$ npm -v # at least:
2.14.2
$ # We recommend using nvm to manage your versions of node and npm.
$ npm install -g gulp
$ npm install
$ gulp
``` 
```
which renders as:

```bash
$ git clone <your fork>
$ node -v # at least:
v4.0.0
$ npm -v # at least:
2.14.2
$ # We recommend using nvm to manage your versions of node and npm.
$ npm install -g gulp
$ npm install
$ gulp
```

BTW, FYI, I'm running node 5.1.0 and npm 3.7.5

**Comment by Dwaynekj** [comment]
Ok will make all the updates tomorrow and merge

**Pull request merged**